src/core/Bot.py

```diff
@@ -243,12 +269,16 @@ async def get_or_fetch_member(self, guild: discord.Guild, member_id: int) -> Opt
             else:
                 return member
 
-        members = await guild.query_members(limit=1, user_ids=(member_id,), cache=True)
+        members = await guild.query_members(limit=1, user_ids=[member_id], cache=True)
 
         if len(members) > 0:
             return members[0]
 
-    async def resolve_member_ids(self, guild: discord.Guild, member_ids: Iterable[int]) -> AsyncIterator[discord.Member]:
+        return None
+
+    async def resolve_member_ids(
+        self, guild: discord.Guild, member_ids: Iterable[int]
+    ) -> AsyncGenerator[discord.Member, None]:
```

Comment:
There is slight difference between `Iterator` and `Generator`

| Iterator | Generator |
|---|---|
| Class is used to implement an iterator  | Function is used to implement a generator.  |
| Iterators are used mostly to iterate or convert other objects to an iterator using `iter()` function.  | Generators are mostly used in loops to generate an iterator by returning all the values in the loop without affecting the iteration of the loop  |
| Every iterator is not a generator  | Every generator is an iterator  |